### PR TITLE
MGMT-2232 Retrieve host HW requirements from backend

### PR DESCRIPTION
**Pull request by mareklibra**

### CODE REVIEW

**Comment by mareklibra** [comment]
hwReqs

File: src/api/utils.ts [hunk 1]
```diff
@@ -51,3 +52,34 @@ export const stringToJSON = <T>(string: string | undefined): T | undefined => {
...
 export const removeProtocolFromURL = (url = '') => url.replace(/^(http|https):\/\//, '');
+
+type FetchApiEndpointParams<T> = {
+  apiCallback: () => AxiosPromise<T>;
+  onSuccess: (data: T) => void;
+  errorTitle: string;
+  onError?: (e: AxiosError<T>) => void;
+};
+
+export const fetchApiEndpoint = <T>({
```
Comment (mareklibra) [inline src/api/utils.ts:63]: I plan to refactor similar code at other places to use this function.

File: src/components/clusterConfiguration/BaremetalInventory.tsx [hunk 2]
```diff
@@ -1,19 +1,30 @@
...
 
 interface BareMetalInventoryProps {
   cluster: Cluster;
 }
 
 const BaremetalInventory: React.FC<BareMetalInventoryProps> = ({ cluster }) => {
   const [isDiscoveryHintModalOpen, setDiscoveryHintModalOpen] = React.useState(false);
+  const [hostRequirements, setHostRequirements] = React.useState<HostRequirements | null>(null);
+
+  React.useEffect(() => {
```
Comment (mareklibra) [inline src/components/clusterConfiguration/BaremetalInventory.tsx:21]: I was thinking about placing this fetch at higher level in the hierarchy but so far there is no other use for it.

**Review by jtomasek — CHANGES_REQUESTED** [review]
Can we maybe postpone introducing `fetchApiEndpoint` helper into a separate PR which would introduce it's usage on multiple places to see how it works out?

File: src/api/utils.ts (hunk 1 above)
Comment (mareklibra) [reply src/api/utils.ts:63]: It's not a must to have it. The pattern repeats, so the idea was to abstract it. I understand your point, I am not big fan of over-abstracting either. So let's postpone such refactoring to a follow-up.

Comment (mareklibra) [reply src/api/utils.ts]: removed from here

File: src/components/clusterConfiguration/BaremetalInventory.tsx [hunk 3]
```diff
@@ -1,19 +1,30 @@
...
+    fetchApiEndpoint<HostRequirements>({
+      apiCallback: getHostRequirements,
+      onSuccess: setHostRequirements,
+      errorTitle: 'Failed to retrieve minimal host requierements',
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply src/components/clusterConfiguration/BaremetalInventory.tsx]: Just out of consistency :-), but both are valid: https://english.stackexchange.com/questions/277236/what-is-the-difference-between-minimum-and-minimal

**Review by jtomasek — APPROVED** [review]

**Pull request merged**